AGENTS.md

```diff
@@ -0,0 +1,304 @@
+# Skyflow Node SDK - AI Coding Agent Instructions
+
+## Project Overview
+
+The Skyflow Node SDK is a TypeScript-based client library for Skyflow's data privacy vault, supporting Node.js, Deno, Bun, and Cloudflare Workers. It provides three primary API surfaces: **Vault** (data operations), **Detect** (PII detection/masking), and **Connections** (integration gateway).
+
+**Key Architecture:**
+- **Client Layer** (`src/vault/client/`): Manages authentication, API initialization, and HTTP configuration
+- **Controller Layer** (`src/vault/controller/`): Business logic for vault, detect, and connection operations
+- **Model Layer** (`src/vault/model/`): Request/Response/Options classes for type-safe API interactions
+- **Generated Code** (`src/_generated_/`): Auto-generated REST client (do not modify manually)
+
+## Request/Response Pattern
+
+All API operations follow a consistent pattern using dedicated classes:
+
+```typescript
+// 1. Create a Request object
+const insertReq = new InsertRequest('table_name', [{ field: 'value' }]);
+
+// 2. Create Options object (if needed)
+const insertOptions = new InsertOptions();
+insertOptions.setReturnTokens(true);
+
+// 3. Call the operation through the client
+const response: InsertResponse = await skyflowClient
+  .vault(vaultId)
+  .insert(insertReq, insertOptions);
+```
+
+**Key Classes:**
+- Request classes: `InsertRequest`, `GetRequest`, `DetokenizeRequest`, `UpdateRequest`, etc.
+- Options classes: `InsertOptions`, `GetOptions`, `DetokenizeOptions` (use setters, not direct property access)
+- Response classes: `InsertResponse`, `GetResponse`, `DetokenizeResponse` (readonly data containers)
+
+## Authentication & Credentials
+
+The SDK supports five credential types (see `src/vault/config/credentials/index.ts`):
+
+1. **Bearer Token** (recommended): `{ token: 'token' }`
+3. **Path to JSON**: `{ path: '/path/to/creds.json' }`
+4. **Credentials String**: `{ credentialsString: JSON.stringify(creds) }`
+5. **Environment Variable**: Auto-reads `SKYFLOW_CREDENTIALS` if no credentials provided
+2. **API Key**: `{ apiKey: 'key' }`
+
+**Credentials Hierarchy:**
+- Individual vault/connection credentials override common credentials
+- Common credentials (`skyflowCredentials`) apply to all vaults/connections
+- Use `validateSkyflowCredentials()` from `src/utils/validations/` for validation
+
+## Client Initialization
+
+The `Skyflow` class (in `src/vault/skyflow/index.ts`) manages multiple vaults and connections:
+
+```typescript
+const skyflowConfig: SkyflowConfig = {
+  vaultConfigs: [vaultConfig1, vaultConfig2],
+  connectionConfigs: [connectionConfig],
+  skyflowCredentials: credentials, // Optional common credentials
```

Comment:
Clarify what's needed if skyflowCredentials is not included. In that case it looks for the ENV variable in the local environment (referenced in the auth section). OR, just don't call it out as optional.